pkg/upgrade/patch/cspc.go

```diff
@@ -59,7 +59,7 @@ func (c *CSPC) PreChecks(from, to string) error {
 		return errors.Errorf("nil cspc object")
 	}
 	version := strings.Split(c.Object.VersionDetails.Status.Current, "-")[0]
-	if version != from && version != to {
+	if version != strings.Split(from, "-")[0] && version != strings.Split(to, "-")[0] {
```

Comment:
We are returning the error with version value after splitting it, we should send the error with the actual value and not after splitting.